### ISSUE TITLE
link-in-text-block is inconsistent with < br> elements

### ISSUE THREAD

**Issue by TobiasG95** [dequelabs/axe-core#4255]
### Product

axe-core

### Product Version

4.8.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
The link-in-text-block issue is only raised when the link is included in the text directly. Using an <br> as a seperator should consistently prevent this issue from being raised.

#### Actual
On some occasions when the < br> is included in a < a> tag then the issue is raised. If the < br> is placed in front of the < a> tag then it is not raised.

#### How to Reproduce
Error noticed using the standalone js module on the cards on this page: https://www.stmuv.bayern.de/themen/index.htm

These are the links:
![image](https://github.com/dequelabs/axe-core/assets/54577126/2277bf3e-da5f-4b9e-9bee-a453457e9ed7)

The link in the left card is not being detected as an issue. This is the html:
![image](https://github.com/dequelabs/axe-core/assets/54577126/d1a51d71-11f7-49bc-9eb4-874cddb506bf)
`<p>Aktuelle Problemlagen untersucht das Bayerische Umweltministerium anhand fundierter wissenschaftlicher Erkenntnisse.<br>
<a title="Informationen zu Forschungsprojekten im Geschäftsbereich des Bayerischen Staatsministeriums für Umwelt und Verbraucherschutz" href="forschung/index.htm">
<img class="bild_doppelpfeil" src="/style/layout/doppelpfeil.gif" alt="" height="8" width="9">
 Forschung</a></p>`

The code on the right is being flagged as an issue:
![image](https://github.com/dequelabs/axe-core/assets/54577126/614abfc7-19f1-45d3-b8ca-5e09b411202d)

The only difference is the placement of the < br> element. It should be consistently either an error or no error. Visually both options look the same.

Let mw know if you have any questions! :-)

**Comment by straker** [comment]
Thanks for the issue. I couldn't reproduce the issue with the page provided. I had to manually edit the HTML of the page to match the screenshots, at which point I was able to reproduce. 

For simplicity here's a reproducible test case:

```html
<div style="color: #000">
  <p>
    Hello world <a href="#" id="br-in" style="color: #333; text-decoration: none;"><br/>here we go</a>
  </p>
  <p>
    Hello world <br/><a href="#" id="br-out" style="color: #333; text-decoration: none;">here we go</a>
  </p>
</div>
```

The first paragraph with the `<br/>` tag inside the link shows a violation, whereas the second paragraph with the `<br/>` outside the link passes.

**Comment by TobiasG95** [comment]
Thanks for the quick feedback! I believe the website owner changed the html to hide the issue.

**Comment by goodwitch** [comment]
I just tested this (using a code pen and the code that straker provided in his comment on 11/29/2023)...and I think this issue is fixed.  Can someone else double-check me...and if you too see it is fixed...can you close this issue?

**Comment by WilcoFiers** [comment]
Tested. This is still an issue.

**Comment by chutchins25** [comment]
**Still reproduces as of 2026-07-21** on `develop` (`57cfe0a3`).

Worth noting: `isInTextBlock` (the algorithm behind this rule) was substantially rewritten in #5000 (merged 2026-03-26), but that rework did not resolve this inconsistency.

Using straker's minimal repro:

```html
<div style="color: #000">
  <p>
    Hello world <a href="#" id="br-in" style="color: #333; text-decoration: none;"><br/>here we go</a>
  </p>
  <p>
    Hello world <br/><a href="#" id="br-out" style="color: #333; text-decoration: none;">here we go</a>
  </p>
</div>
```

Running `link-in-text-block` on this:

```js
const results = await axe.run(fixture, {
  runOnly: { type: 'rule', values: ['link-in-text-block'] }
});
// violations: ["#br-in"]
// #br-out: absent from violations, passes, and incomplete (rule doesn't apply)
```

The two visually-identical links land in different states: `#br-in` (the `<br>` inside the link) is flagged as a violation, while `#br-out` (the `<br>` before the link) is skipped entirely. The underlying check confirms the split — `isInTextBlock(#br-in) === true` but `isInTextBlock(#br-out) === false`.
